### ISSUE TITLE
Apprise Integration

### ISSUE THREAD

**Issue by conor-f** [go-vikunja/vikunja#8]
[Apprise](https://github.com/caronc/apprise) is a platform-rich notifier. I think it would be a huge addition to notifications on Vikunja. It is a Python package which would make it awkward for integrating, but it also exposes a CLI util which could make this easier if you wanted to go down that route.

I have no experience with Go so I'm just filing this as a feature request and not accompanying it with a PR!

**Comment by georgejung** [comment]
I just discovered apprise and am trying to connect it to everything I can. An integration into vikunja would be great!

**Comment by typkrft** [comment]
This could be done more generically with some sort of outgoing webhook integration. This would allow a lot of complexity for users without adding a lot of complexity to the code base. Maybe a selection of events, the ability to add specific headers and a url. This would allow cool integrations with things like ntfy, appraise, and n8n.

**Comment by kolaente** [comment]
@typkrft do you think it would help if there was a webhook for notifications?

**Comment by typkrft** [comment]
Yes, an outgoing webhook similar to what you already have works well already. I setup up n8n to parse the current outgoing webhook and then send it to ntfy after I posted my comment. It works great. A more direct way to do it would be ideal. Really the only thing that would need to be changed from what you have is the ability to add headers and some templating, that way people can use it with whatever service they'd like. This could be done in the same project webhook popup where you select the events so you can use different webhooks per project. Obviously you could hook it into whatever you want, or implement it in any way, but I'm just trying to think of the easiest way to satisfy everyone. 

`url:` `https://ntfy.your.server`
`headers:` `{"someJson": "...."}`
`body:` `Project: {{ project.name }} - ...`

Something like that would work for appraise, ntfy, etc.

**Comment by kolaente** [comment]
I like the idea of templates for webhooks. To solve this issue here, we'd need to do these two things:

1. Add a "personal" webhook which fires every time a notification happens
2. Add templating to webhooks, so that it is possible to use a wide variety of notification services. Possibly with presets for ntfy, apprise, etc.

**Comment by RileyMathews** [comment]
I just found this thread after searching if vikunja had a Ntfy integration. I have been using vikunja for the last year in my homelab for kanban and have been loving it so much I was hoping it could replace apple reminders for me and started looking into its notifications features. The ability to do what you described with webhook templating along with a reminders webhook would be absolutely amazing and let me use vikunja for so many workflows! So huge +1 to this idea.

I don’t have any real experience with go but if you are not planning to pick this up soon I might try to find some free time to see if this is something I could contribute.

**Comment by kolaente** [comment]
Would love to get a contribution for webhook templates, it's not likely for me to start working on this soon.

**Comment by RileyMathews** [comment]
I had some time to setup my devenv today and things are up and running pretty smoothly. I think my first thing I will try to contribute is just a simple hardcoded webhook for when reminders are sent instead of just having them sent via email. Then depending on how well I take to Go I may take that to the next step and see about web hook templates.

**Comment by kolaente** [comment]
@RileyMathews awesome, looking forward to it! It's probably best if you extend the notification abstraction with another option for webhooks, similar to how we currently have it for mail and database.

Feel free to ping me here or on matrix if you have any specific questions about a possible implementation.

**Comment by Spielzeugpferd** [comment]
I don't know if still relevant but the agent and myself created two small plugins that listen to vikunja's notification events and exposes them towards your local [apprise](https://github.com/Spielzeugpferd/vikunja-apprise-plugin) or [ntfy](https://github.com/Spielzeugpferd/vikunja-ntfy-plugin) server.
